bigtable/tests/admin_integration_test.cc

```diff
@@ -12,185 +12,269 @@
+                   back_inserter(diffTableList));
+
+    return diffTableList.empty();
   }
-  std::sort(actual.begin(), actual.end());
-  std::sort(expected.begin(), expected.end());
-  std::vector<std::string> differences;
-  std::set_difference(expected.begin(), expected.end(), actual.begin(),
-                      actual.end(), std::back_inserter(differences));
-  if (differences.empty()) {
-    std::cout << "ListTables() successful" << std::endl;
-    return;
+
+  bool checkTableSchema(admin_proto::Table const& actualTable,
+                        string const& expectedText, string const& message) {
+    admin_proto::Table expectedTable;
+
+    // ASSERT_FALSE(google::protobuf::TextFormat::ParseFromString(expectedText,
+    // &expectedTable));
+    if (not google::protobuf::TextFormat::ParseFromString(expectedText,
+                                                          &expectedTable))
+      return false;
+
+    string delta;
+    google::protobuf::util::MessageDifferencer msgDifferencer;
+    msgDifferencer.ReportDifferencesToString(&delta);
+    // ASSERT_FALSE(msgDifferencer.Compare(expectedTable, actualTable));
+    if (not msgDifferencer.Compare(expectedTable, actualTable)) return false;
+
+    return true;
   }
-  std::ostringstream os;
-  os << "Mismatched names: " << absl::StrJoin(differences, "\n")
-     << "\nactual: " << absl::StrJoin(actual, "\n")
-     << "\nexpected: " << absl::StrJoin(expected, "\n") << "\n";
-  throw std::runtime_error(os.str());
+};
+}
+}
+
+namespace bigtable {
+namespace testing {
+
+/**
+ *  Test case for checking if Instance is empty or not.
+ *  If instance is empty then go ahead.
+ *  If instance is not empty the throw assertion
+ * */
+TEST_F(AdminIntegrationTest, CheckInstanceIsEmpty) {
+  namespace admin_proto = ::google::bigtable::admin::v2;
+
+  ASSERT_TRUE(table_admin_->ListTables(admin_proto::Table::NAME_ONLY).empty());
+}
+
+/***
+ * Test case for checking create table
+ * If created tableID and passed tableID is same then test is successful.
+ *
+ * */
+TEST_F(AdminIntegrationTest, CheckCreateTable) {
+  string table_id = "table0";
+
+  // Create Table
+  auto table = table_admin_->CreateTable(table_id, bigtable::TableConfig());
+
+  // Check table is created properly
+  auto table_result = table_admin_->GetTable(table_id);
+
+  ASSERT_EQ(table.name(), table_result.name());
+
+  // Delete this table so that next run should not throw error
+  table_admin_->DeleteTable(table_id);
+}
+
+/**
+ * Check if list of table names are matching with the
+ * expected tablename list
+ *
+ * */
+
+TEST_F(AdminIntegrationTest, CheckTableListWithSingleTable) {
+  const string table_id = "table0";
+
+  // Create table first here.
+  table_admin_->CreateTable(table_id, bigtable::TableConfig());
+
+  vector<string> expectedTableList = {
+      absl::StrCat(table_admin_->instance_name(), "/tables/", table_id)};
+
+  bool tableListIsEmpty = TestForTableListCheck(expectedTableList);
```

Comment:
Please following the naming conventions for local variables, this should be `table_list_is_empty`, or better yet: `instance_is_empty` because that tells us what the table list represents.